lib/Checksum.py

```diff
@@ -0,0 +1,102 @@
+import base64
+import string
+import random
+import hashlib
+
+from Crypto.Cipher import AES
+
+IV = "@@@@&&&&####$$$$"
+BLOCK_SIZE = 16
+
+
+def generate_checksum(param_dict, merchant_key, salt=None):
+    params_string = __get_param_string__(param_dict)
+    salt = salt if salt else __id_generator__(4)
+    final_string = '%s|%s' % (params_string, salt)
+
+    hasher = hashlib.sha256(final_string.encode())
+    hash_string = hasher.hexdigest()
+
+    hash_string += salt
+
+    return __encode__(hash_string, IV, merchant_key)
+
+
+def generate_checksum_by_str(param_str, merchant_key, salt=None):
+    params_string = param_str
+    salt = salt if salt else __id_generator__(4)
```

Comment:
What is salt here? Add a comment or improve the naming convention.

Reply:
A cryptographic salt is an additional input other than the message itself for a hash function so that it prevents an attacker from launching dictionary attacks.